qubes/api/admin.py

```diff
@@ -336,6 +354,49 @@ def vm_volume_import(self):
 
         return '{} {}'.format(size, path)
 
+    @qubes.api.method('admin.vm.tag.List', no_payload=True)
+    @asyncio.coroutine
+    def vm_tag_list(self):
+        assert not self.arg
+
+        tags = self.dest.tags
+
+        tags = self.fire_event_for_filter(tags)
+
+        return ''.join('{}\n'.format(tag) for tag in sorted(tags))
+
+    @qubes.api.method('admin.vm.tag.Get', no_payload=True)
+    @asyncio.coroutine
+    def vm_tag_get(self):
+        qubes.vm.Tags.validate_tag(self.arg)
+
+        self.fire_event_for_permission()
+
+        return '1' if self.arg in self.dest.tags else '0'
```

Comment:
> The only string that's False is an empty string. As to the value, we could either use '1', since it's non-empty and clearly "true", or full tag name.

One (of) issues with this (and also using empty string as "false" value for features) is that if something fails and response is truncated (either inside qubesd, or in transit), it will be mistaken for "false" value. This is one reason why it's better to have defined _non empty_ values for for valid responses. Another reason is that some languages have hardly distinction between "no value" and "empty value" (think: shell). This API should be designed to make mistakes harder (this is not the contest who makes bigger footgun!). Management VM will make security-critical decisions based on API responses!

> Do we allow zero-length tag name?

No. And also we allow only very limited character set in tag names, even though parts used in current Admin API implementation would not have troubles with full ASCII or even unicode. Exactly for the reason explained above.